src/app/(main)/todos/[todoId]/note/create/page.tsx

```diff
@@ -10,25 +13,114 @@ import LinkAttached from "@/components/note/link-attached";
 import LoadNoteToastManager, {
   NoteData,
 } from "@/components/note/load-note-toast-manager";
-import TagInput from "@/components/note/tag";
+import TagInput, { Tag } from "@/components/note/tag";
 import TempSaveManager, {
   TempSaveManagerRef,
 } from "@/components/note/temp-save-manager";
 import Tiptap from "@/components/note/tiptap";
 import Todo from "@/components/note/todo";
+import { createNote, getNote, updateNote } from "@/services/note";
 import "@/styles/note.css";
+import { Note } from "@/types/note";
 import { countWithoutSpaces, countWithSpaces } from "@/utils/text-utils";
 
 export default function NotesPage() {
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const params = useParams();
+  const noteIdParam = searchParams.get("noteId");
+  // noteId가 있으면 수정모드, 없으면 새로 작성하는 모드
+  const noteId = noteIdParam ? Number(noteIdParam) : null;
+  // [todoId]는 라우트 동적 세그먼트에서 추출
+  const todoId = params?.todoId ? Number(params.todoId) : null;
+  const isEditMode = noteId !== null;
```

Comment:
useParams를 쓰지 않고 아래처럼 쓰는 건 어떠신가요? Next.js 공식문서에서 동적 경로에서 값을 받아오는 방법을 소개할 때 아래 방식을 주로 쓰더라구요.

```suggestion
export default function NotesPage({ params }: { params: { todoId: string } }) {
  const router = useRouter();
  const searchParams = useSearchParams();
  const noteIdParam = searchParams.get("noteId");
  // noteId가 있으면 수정모드, 없으면 새로 작성하는 모드
  const noteId = noteIdParam ? Number(noteIdParam) : null;
  // [todoId]는 라우트 동적 세그먼트에서 추출
  const todoId = Number(params.todoId);
  const isEditMode = noteId !== null;
```